editor/src/components/navigator/navigator-item/navigator-condensed-entry.tsx

```diff
@@ -94,9 +119,13 @@ export const CondensedEntryItemWrapper = React.memo(
           display: 'flex',
           alignItems: 'center',
           backgroundColor: rowRootSelected
-            ? colorTheme.selectionBlue.value
+            ? isComponentOrInsideComponent
+              ? colorTheme.selectionPurple.value
+              : colorTheme.selectionBlue.value
             : hasSelection || wholeRowInsideSelection
-            ? colorTheme.childSelectionBlue.value
+            ? isComponentOrInsideComponent
+              ? colorTheme.childSelectionPurple.value
+              : colorTheme.childSelectionBlue.value
```

Comment:
feels like these three lines should be somehow turned into a helper, because they are repeated many times across the navigator

Reply:
oh sure, makes sense. Done in https://github.com/concrete-utopia/utopia/pull/5922/commits/1ea36756eba76758d0f56e61dc642aac44f944e6